src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs

```diff
@@ -159,12 +158,17 @@ public virtual async Task<ProjectPackages> GetInstalledAndTransitivePackagesAsyn
 
             CounterfactualLoggers.TransitiveDependencies.EmitIfNeeded(); // Emit only one event per VS session
             IEnumerable<TransitivePackageReference> transitivePackagesWithOrigins;
-            if (await ExperimentUtility.IsTransitiveOriginExpEnabled.GetValueAsync(token))
+            if (await ExperimentUtility.IsTransitiveOriginExpEnabled.GetValueAsync(token) && useTransitiveOrigins)
             {
                 // Compute Transitive Origins
                 Dictionary<string, TransitiveEntry> transitiveOrigins;
                 if (IsInstalledAndTransitiveComputationNeeded || TransitiveOriginsCache == null)
                 {
+                    if (targetsList == null) // special case
```

Comment:
Could you please describe the special case as part of the comment?

Reply:
Added comment

Reply:
The comment doesn't quite match the logic. Based on the comment I would expect to check (IsInstalledAndTransitiveComputationNeeded && TransitiveOriginsCache == null). Am I missing something?